AdyenComponents/BACS Direct Debit/BACSDirectDebitComponent.swift

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Adyen
+import UIKit
+
+internal protocol BACSDirectDebitRouterProtocol {
+    func presentConfirmation(with data: BACSDirectDebitData)
+    func confirmPayment(with data: BACSDirectDebitData)
+    func cancelPayment()
+}
+
+/// A component that provides a form for BACS Direct Debit payments.
+public final class BACSDirectDebitComponent: PaymentComponent, PresentableComponent {
+
+    // MARK: - PresentableComponent
+
+    public var viewController: UIViewController
+
+    public var requiresModalPresentation: Bool = false
+
+    /// The object that acts as the delegate of the component.
+    public weak var delegate: PaymentComponentDelegate?
+
+    /// The BACS Direct Debit payment method.
+    public let paymentMethod: PaymentMethod
+
+    /// :nodoc:
+    public let apiContext: APIContext
+
+    /// :nodoc:
+    public let style: FormComponentStyle
+
+    /// :nodoc:
+    public var localizationParameters: LocalizationParameters?
+
+    // MARK: - Initializers
+
+    /// Creates and returns a BACS Direct Debit component.
+    /// - Parameters:
+    ///   - paymentMethod: The BACS Direct Debit payment method.
+    ///   - apiContext: The API context.
+    ///   - style: The component's UI style.
+    ///   - localizationParameters: The localization parameters.
+    public init(paymentMethod: BACSDirectDebitPaymentMethod,
+                apiContext: APIContext,
+                style: FormComponentStyle = .init(),
+                localizationParameters: LocalizationParameters? = nil) {
+        self.paymentMethod = paymentMethod
+        self.apiContext = apiContext
+        self.style = style
+        self.localizationParameters = localizationParameters
+
+        let view = BACSDirectDebitInputFormViewController(title: paymentMethod.name,
+                                                          styleProvider: style)
+        self.viewController = view as UIViewController
+
+        let itemsFactory = BACSDirectDebitItemsFactory(styleProvider: style,
+                                                       localizationParameters: localizationParameters,
+                                                       scope: self)
```

Comment:
I see `scope` is retained by `BACSDirectDebitItemsFactory`, and `itemsFactory` is retained by the `presenter` and presenter is retained by self which is the `scope`, this is a retain cycle, please make the `scope` weak in `BACSDirectDebitItemsFactory`.

Reply:
`scope` could be just String.

Reply:
`scope` (BACSComponent) doesn't retain the items factory, only used on init but it doesn't keep a reference to it.

I was checking the dependencies. Since the `BACSView` is weakly retaining `BACSPresenter`, there's no retain cycle. Here's a diagram for better clarity.

![Screen Shot 2021-12-06 at 11 38 56](https://user-images.githubusercontent.com/10688366/144832195-73bda007-e347-4591-be2d-1387e806a0e2.png)

Despite all this, I agree with @descorp. I'm gonna use scope as String instead. What do you think?

Reply:
Used String instead